client/signup/steps/site-type/form.jsx

```diff
@@ -44,29 +52,88 @@ class SiteTypeForm extends Component {
 		this.props.submitForm( type );
 	};
 
-	renderSegmentOptions() {
-		return getAllSiteTypes().map( siteTypeProperties => (
-			<Card
-				className="site-type__option"
-				key={ siteTypeProperties.id }
-				displayAsLink
-				data-e2e-title={ siteTypeProperties.slug }
-				tagName="button"
-				onClick={ this.handleSubmit.bind( this, siteTypeProperties.slug ) }
-			>
-				<strong className="site-type__option-label">{ siteTypeProperties.label }</strong>
-				<span className="site-type__option-description">{ siteTypeProperties.description }</span>
+	handleSubmitOther = () => {
+		if ( ! this.state.otherValue || this.state.hasOtherReasonFocus ) {
+			return;
+		}
+		this.handleSubmit( 'other—' + this.state.otherValue );
+	};
+
+	setOtherReasonFocus = focus => () => {
+		this.setState( { hasOtherReasonFocus: focus } );
+	};
+
+	renderBasicCard = () => {
+		return (
+			<Card className="site-type__wrapper">
+				{ getAllSiteTypes().map( siteTypeProperties => (
+					<Card
+						className="site-type__option"
+						key={ siteTypeProperties.id }
+						displayAsLink
+						data-e2e-title={ siteTypeProperties.slug }
+						onClick={ this.handleSubmit.bind( this, siteTypeProperties.slug ) }
+					>
+						<strong className="site-type__option-label">{ siteTypeProperties.label }</strong>
+					</Card>
+				) ) }
 			</Card>
-		) );
-	}
+		);
+	};
+
+	renderOtherInfo = () => {
+		const { translate } = this.props;
+
+		return (
+			<div className="site-type__other-option">
+				<p className="site-type__other-label">{ translate( 'Or type your own' ) }</p>
+
+				<div className="site-type__other-form">
+					<FormTextInput
+						className="site-type__other-input"
+						type="text"
+						selectOnFocus
+						placeholder={ translate( 'Other' ) }
+						onChange={ this.onOtherCatChange }
+						value={ this.state.otherValue }
+					/>
+
+					<Button
+						className="site-type__other-submit"
+						disabled={ false }
+						onClick={ this.handleSubmitOther }
+					>
+						{ translate( 'Continue' ) }
+					</Button>
+				</div>
+			</div>
+		);
+	};
 
 	render() {
-		return <Card className="site-type__wrapper">{ this.renderSegmentOptions() }</Card>;
+		const { isJetpack } = this.props;
+
+		if ( isJetpack === null ) {
```

Comment:
This will break if this step eventually gets displayed on a route with a site slug for a non-Jetpack site. In that case, `isJetpack` will be `false`. So I suggest making this a simple bool check to go around this case.

```suggestion
		if ( ! isJetpack ) {
```